integration/class-surge.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * Surge integration file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Integration;
+
+use function Activitypub\user_can_activitypub;
+
+/**
+ * Surge Cache integration.
+ *
+ * This class handles the compatibility with the Surge plugin.
+ *
+ * @see https://wordpress.org/plugins/surge/
+ */
+class Surge {
+	/**
+	 * The define for the Surge cache config.
+	 *
+	 * @var string
+	 */
+	public static $cache_config = 'define( \'WP_CACHE_CONFIG\', \'' . ACTIVITYPUB_PLUGIN_DIR . 'integration/surge-cache-config.php\' );';
+
+	/**
+	 * The pattern to find the define for the Surge cache config.
+	 *
+	 * @var string
+	 */
+	public static $cache_config_pattern = '/define\s*\(\s*[\'"](WP_CACHE_CONFIG)[\'"]\s*,\s*[\'"](.*?)[\'"]\s*\)\s*;/i';
+
+	/**
+	 * Initialize the Surge integration.
+	 */
+	public static function init() {
+		\add_action( 'activate_surge/surge.php', array( self::class, 'add_cache_config' ) );
+		\add_action( 'deactivate_surge/surge.php', array( self::class, 'remove_cache_config' ) );
+
+		\add_filter( 'site_status_tests', array( self::class, 'maybe_add_site_health' ) );
+	}
+
+	/**
+	 * Add the Surge cache config.
+	 */
+	public static function add_cache_config() {
+		$file = self::get_config_file_path();
+
+		if ( ! \wp_is_writable( $file ) ) {
+			return;
+		}
+
+		if ( ! \function_exists( 'WP_Filesystem' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+		}
+
+		global $wp_filesystem;
+		\WP_Filesystem();
+
+		$config = $wp_filesystem->get_contents( $file );
+
+		// Check if the define already exists.
```

Comment:
```suggestion
		// Check if the constant already exists.
```
My IDE complained about grammar on this one? "constant" fixes it.